src/kv/key.rs

```diff
@@ -57,7 +57,7 @@ impl<'k> fmt::Display for Key<'k> {
 }
 
 impl<'k> AsRef<str> for Key<'k> {
-    fn as_ref(&self) -> &str {
+    fn as_ref(&self) -> &'k str {
```

Comment:
This has no effect unless you add `#[refine]` (https://rust-lang.github.io/rfcs/3245-refined-impls.html).

Reply:
Hmm, it almost seems weird this compiles at all :eyes: I hadn't seen `#[refine]` before. Thanks for pointing it out!